### PR TITLE
feat(core): expose projectCwd to the script env

### PR DESCRIPTION
**Pull request by elmpp**
**What's the problem this PR addresses?**

Having ready access to the base of the project within a script entry. 

**How did you fix it?**

Add to the scriptEnv within the shell setup

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: .yarn/versions/de729474.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  - "@yarnpkg/plugin-patch"
+  - "@yarnpkg/plugin-pnp"
+  - "@yarnpkg/plugin-stage"
+  - "@yarnpkg/plugin-typescript"
+  - "@yarnpkg/plugin-version"
+  - "@yarnpkg/plugin-workspace-tools"
+  - "@yarnpkg/builder"
+  - "@yarnpkg/cli"
+  - "@yarnpkg/doctor"
+  - "@yarnpkg/pnpify"
```
Comment (elmpp) [inline .yarn/versions/de729474.yml:30]: hope this is ok - wasn't sure how docs are released, for instance

File: .yarn/versions/de729474.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+releases:
+  "@yarnpkg/core": patch
```
Comment (merceyz) [inline .yarn/versions/de729474.yml:2]: This is technically a feature (minor) and it needs a release of the CLI as well

**Comment by arcanis** [comment]
Thanks! I've added them this time, but remember that tests are helpful to prevent accidental regressions 😉

**Comment by jamiter** [comment]
FYI, this has not been added to the changelog, so it's unclear when it will be released. 

(Looking forward to the release 🤞😉)

**Pull request merged**